gbus/tx/mysql/timeout.go

```diff
@@ -0,0 +1,203 @@
+package mysql
+
+import (
+	"database/sql"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/wework/grabbit/gbus"
+)
+
+var _ gbus.TimeoutManager = &TimeoutManager{}
+
+//TimeoutManager is a mysql implementation of a persistent timeoutmanager
+type TimeoutManager struct {
+	Bus         gbus.Bus
+	Log         func() logrus.FieldLogger
+	TimeoutSaga func(*sql.Tx, string) error
+	Txp         gbus.TxProvider
+	SvcName     string
+	exit        chan bool
+}
+
+func (tm *TimeoutManager) ensureSchema() error {
+	tblName := tm.GetTimeoutsTableName()
+	tx, e := tm.Txp.New()
+	if e != nil {
+		tm.Log().WithError(e).Error("failed to create schema for mysql timeout manager")
+		return e
+	}
+
+	createTableSQL := `CREATE TABLE IF NOT EXISTS ` + tblName + ` (
+      rec_id INT PRIMARY KEY AUTO_INCREMENT,
+      saga_id VARCHAR(255) UNIQUE NOT NULL,
+	  timeout DATETIME NOT NULL,
+	  INDEX ix_` + tm.GetTimeoutsTableName() + `_timeout_date(timeout)
+      )`
+
+	if _, e := tx.Exec(createTableSQL); e != nil {
+		if rbkErr := tx.Rollback(); rbkErr != nil {
+			tm.Log().Warn("timeout manager failed to rollback transaction")
+		}
+		return e
+	}
+	return tx.Commit()
+}
+
+func (tm *TimeoutManager) purge() error {
+	purgeSQL := `DELETE FROM ` + tm.GetTimeoutsTableName()
```

Comment:
Truncate table would be better

Reply:
it know but it causes me problems for some reason when I run the tests...so i guess delete will be fine